docs/streams/upgrade-guide.html

```diff
@@ -76,8 +76,13 @@ <h3><a id="streams_api_changes_240" href="#streams_api_changes_240">Streams API
 
     <!-- Placeholder KIP-213 -->
     <!-- Placeholder KIP-307 -->
-    <!-- Placeholder KIP-479 -->
+    <p>
+        Before the 2.4.0 version of Kafa Streams, users of the DSL could not name the state stores involved in a Stream-Stream join.  If users added DSL operations upstream of the 
+        join, the internal names of the state stores would shift, requiring an application reset when redeploying.  Now in the 2.4.0 release Kafka Streams adds the <code>StreamJoined</code>
+        object, which gives users the ability to name the join processor, repartition topic(s) (if a repartition is required), and the state stores involved in the join.  Also, by naming the state stores, the changelog topics backing the state stores are named as well.
 
+        Another feature delivered by <code>StreamJoined</code> that will be of interest to users is that you can now configure the type of state store used in the join.  Users can now elect to use in-memory stores or custom state stores for a Stream-Stream join.  Users should note that the provided stores will not be available for searching via Interactive Queries.  With the addition of <code>StreamJoined</code> Stream-Stream join operations using <code>Joined</code> have been deprecated, and users are encouraged to switch over to Stream-Stream join methods using the new overloaded methods.  You can get more details from <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-479%3A+Add+StreamJoined+config+object+to+Join">KIP-479</a>
```

Comment:
nits: (single line so can't refer to specific location)
1) "searching via IQ" -> "querying via IQ"? 
2) The sentence beginning "With the addition..." reads a bit awkwardly to me. Maybe something like "stream-stream joins accepting a Joined parameter have been deprecated; users are encouraged to migrate to the new StreamJoined parameter (is that an accurate description?)
3) I would also maybe refactor the first sentence a bit, something like "With the addition of the StreamJoined object, you can now configure the type of state store..."

Reply:
ack